primitives/src/erasure_code/reed_solomon_erasure.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! Module for Reed Solomon Erasure Code
+
+use crate::errors::PrimitivesError;
+use ark_ff::PrimeField;
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+use ark_std::{borrow::Borrow, string::ToString, vec, vec::Vec};
+use core::marker::PhantomData;
+
+use super::ErasureCode;
+
+/// Very naive implementation of Reed Solomon erasure code.
+///  * `reconstruction_size`: and the minimum number of shards required for
+///    reconstruction
+///  * `num_shards`: the block (codeword) length
+/// The encoding of a message is the evaluation on (1..num_shards) of the
+/// polynomial whose coefficients are the message entries. Decoding is a naive
```

Comment:
As discussed offline, we'll keep data in coefficient form because:
1. It's easier
2. We need the data in coefficient form anyway for use in VID. Thus, evaluation form would require 2 FFTs in `encode`, which is unacceptable in a performance-critical section